File: java/bubble_sort.java
public class bubble_sort {
    public static void main(String[] args) {
        int[] example = {5, 1, 6, 3, 4};
        bubble_sort_(example);
        for (int arg : example) {
            System.out.printf("%d ", arg);
        }
    }

    public static void bubble_sort_(int[] arr) {
        int l = arr.length - 1;
        boolean changes = true;
        while (changes) {
            changes = false;
            for (int ind = 0; ind < l; ind++) {
                if (arr[ind] > arr[ind + 1]) {
                    int temp = arr[ind];
                    arr[ind] = arr[ind + 1];
                    arr[ind + 1] = temp;
                    changes = true;
                }
            }
        } // no need to return the array
    }
}

File: python/bubble_sort.py
def main():
    example = [5, 1, 6, 3, 4]
    bubble_sort(example)
    print(example)

def bubble_sort(arr):
    l = len(arr) - 1
    changes = True
    while changes:
        changes = False
        for ind in range(l):
            if arr[ind] > arr[ind + 1]:
                arr[ind], arr[ind + 1] = arr[ind + 1], arr[ind]
                changes = True
    # no need to return the array

if __name__ == '__main__':
    main()
